### PR TITLE
Fix egress mtls origination test

### PR DESCRIPTION
**Pull request by keithmattix**
This should fix the failing test in istio/istio#50040, but we need to make sure we capture the underlying behavior change in a release/upgrade note somewhere.

### CODE REVIEW

File: content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md [hunk 1]
```diff
@@ -527,6 +527,8 @@ to hold the configuration of the NGINX server:
       namespace: mesh-external
       labels:
         run: my-nginx
+      annotations
+        networking.istio.io/exportTo: "." # simulate a test service by not exporting outside this namespace
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/index.md:531]:
```suggestion
        networking.istio.io/exportTo: "." # simulate an external service by not exporting outside this namespace
```

File: content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/snips.sh [hunk 2]
```diff
@@ -348,6 +348,8 @@ metadata:
   namespace: mesh-external
   labels:
     run: my-nginx
+  annotations
+    networking.istio.io/exportTo: "." # simulate a test service by not exporting outside this namespace
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/snips.sh:352]:
```suggestion
    networking.istio.io/exportTo: "." # simulate an external service by not exporting outside this namespace
```

**Comment by keithmattix** [comment]
@frankbu Any idea why gencheck is failing? I have no diff locally

**Comment by frankbu** [comment]
Seems like a go.mod difference:
```
diff --git a/go.mod b/go.mod
index 40a84f7a8..d0674f050 100644
--- a/go.mod
+++ b/go.mod
@@ -11,11 +11,8 @@ exclude k8s.io/kubernetes v1.13.0
 replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.5
 
 require (
-	github.com/envoyproxy/go-control-plane v0.12.1-0.20240409154308-6a432fea92ca
 	github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2
 	golang.org/x/sync v0.7.0
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20240401170217-c3f982113cda
-	google.golang.org/grpc v1.63.0
 	istio.io/istio v0.0.0-20240413233931-e1eeb10b4f36
 	k8s.io/apimachinery v0.30.0-rc.2
 	k8s.io/client-go v0.30.0-rc.2
@@ -50,6 +47,7 @@ require (
 	github.com/docker/docker v25.0.5+incompatible // indirect
 	github.com/docker/docker-credential-helpers v0.8.1 // indirect
 	github.com/emicklei/go-restful/v3 v3.11.2 // indirect
+	github.com/envoyproxy/go-control-plane v0.12.1-0.20240409154308-6a432fea92ca // indirect
 	github.com/envoyproxy/protoc-gen-validate v1.0.4 // indirect
 	github.com/evanphx/json-patch v5.9.0+incompatible // indirect
 	github.com/evanphx/json-patch/v5 v5.9.0 // indirect
@@ -190,6 +188,8 @@ require (
 	golang.org/x/tools v0.19.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.4.0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20240401170217-c3f982113cda // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240401170217-c3f982113cda // indirect
+	google.golang.org/grpc v1.63.0 // indirect
 	google.golang.org/protobuf v1.33.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
```
You can fix it with:
```
git apply <(curl -sL "https://gcsweb.istio.io/gcs/istio-prow/pr-logs/pull/istio_istio.io/14914/gencheck_istio.io/1780982059813048320/artifacts/check-clean-repo-diff.patch")

**Comment by ericvn** [comment]
Just a note that the `gencheck` failure will be fixed my doing a `make clean gen`. `clean` is needed to remove the test framework files copied in which will update the go.* slightly.

**Comment by ericvn** [comment]
Easiest to fix with a `make clean gen`

**Comment by ericvn** [comment]
/test doc.test.profile-demo

flake

**Comment by kfaseela** [comment]
did we get this documented as well?

File: content/en/docs/tasks/traffic-management/egress/egress-tls-origination/mtls_test.sh [hunk 3]
```diff
@@ -21,7 +21,6 @@ set -e
 set -u
 set -o pipefail
 
-source "content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/snips.sh"
```
Comment (kfaseela) [inline content/en/docs/tasks/traffic-management/egress/egress-tls-origination/mtls_test.sh:24]: why is this not working anymore? is the exportTo part the only change?

File: vm-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: install.istio.io/v1alpha1
```
Comment (zirain) [inline vm-cluster.yaml:1]: nit?

File: content/en/docs/tasks/traffic-management/egress/egress-tls-origination/index.md [hunk 5]
```diff
@@ -217,12 +217,174 @@ TLS origination.
...
+    $ kubectl apply -f - <<EOF
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: my-nginx
+      namespace: mesh-external
+      labels:
+        run: my-nginx
+      annotations:
+        "networking.istio.io/exportTo": "." # simulate an external service by not exporting outside this namespace
```
Comment (zirain) [inline content/en/docs/tasks/traffic-management/egress/egress-tls-origination/index.md:338]: this's the only thing different?

Comment (keithmattix) [reply content/en/docs/tasks/traffic-management/egress/egress-tls-origination/index.md:338]: Substantively yes

File: content/en/docs/tasks/traffic-management/egress/egress-tls-origination/mtls_test.sh (hunk 3 above)
Comment (keithmattix) [reply content/en/docs/tasks/traffic-management/egress/egress-tls-origination/mtls_test.sh:24]: The issue is the the egress gateway mTLS tests used the same setup code but must not use export to. Removing this import breaks the dependency between the workload and gateway mTLS tests

**Comment by keithmattix** [comment]
> did we get this documented as well?

I'm trying to track down where the change occurred and add a release note

File: vm-cluster.yaml (hunk 4 above)
Comment (ericvn) [reply vm-cluster.yaml:1]: I think you can remove this file as it's just something that is left over on your local machine. Looking it seems to be created as part of https://istio.io/latest/docs/setup/install/virtual-machine/ and we don't actually delete it as part of the cleanup.

**Comment by ericvn** [comment]
> I'm trying to track down where the change occurred and add a release note

Did you find/create the release note? I don't think that effort needs to hold this PR out, so once the extraneous file is removed, this is ready for merge?

**Comment by keithmattix** [comment]
@ericvn correct

**Review by ericvn — APPROVED** [review]

**Comment by keithmattix** [comment]
The context can be found in istio/istio#50478

**Comment by keithmattix** [comment]
/test doc.test.profile-demo
Flake?

**Review by kfaseela — APPROVED** [review]

**Comment by ericvn** [comment]
Removed the cherry-pick 1.22 label since we don't branch 1.22 until the actual release of 1.22.0. For now, the master branch reflects what will be released in 1.22.0 (and at preliminary.istio.io).

**Pull request merged**